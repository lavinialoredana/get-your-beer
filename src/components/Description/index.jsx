const Description = ({ descriptionField, handleDescriptionChange }) => {
  return (
    <div className="description-container">
      <textarea
        data-testid="description"
        rows="50"
        cols="4"
        label="description"
        value={descriptionField}
        onChange={handleDescriptionChange}
      />
    </div>
  );
};

export default Description;
