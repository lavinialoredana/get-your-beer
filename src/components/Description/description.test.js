import { render, screen } from "@testing-library/react";
import Description from ".";

describe("description component", () => {
  it("to match snapshot", () => {
    //Arrange
    const { asFragment } = render(
      <Description
        descriptionField="Hello"
        handleDescriptionChange={jest.fn()}
      />
    );
    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
  it("should contain textarea", () => {
    render(
      <Description
        descriptionField="Hello"
        handleDescriptionChange={jest.fn()}
      />
    );
    const descriptionElement = screen.getByTestId("description");
    expect(descriptionElement).toBeInTheDocument;
  });
});
