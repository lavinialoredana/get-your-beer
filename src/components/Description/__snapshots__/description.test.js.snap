// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`description component to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="description-container"
  >
    <textarea
      cols="4"
      data-testid="description"
      label="description"
      rows="50"
    >
      Hello
    </textarea>
  </div>
</DocumentFragment>
`;
